File: src/libs/ThirdParty.CSSuppliers/ExpediaRapid/SerializableClasses/Search/SearchResponse.cs
namespace ThirdParty.CSSuppliers.ExpediaRapid.SerializableClasses.Search
{
    using System.Collections.Generic;
    using Newtonsoft.Json;

    public class SearchResponse : List<PropertyAvailablility>, IExpediaRapidResponse<SearchResponse>
    {
        public (bool valid, SearchResponse response) GetValidResults(string responseString, int statusCode)
        {
            (bool valid, SearchResponse response) = (false, new SearchResponse());
            if (!string.IsNullOrWhiteSpace(responseString))
            {
                try
                {
                    response = JsonConvert.DeserializeObject<SearchResponse>(responseString)!;
                    valid = true;
                }
                catch
                {
                }
            }
            return (valid, response);
        }
    }
}

File: src/libs/ThirdParty.VBSuppliers/Properties/ExpediaRapid/SerializableClasses/Search/SearchResponse.vb
Imports Intuitive.Net
Imports Newtonsoft.Json
Imports Newtonsoft.Json.Linq

Namespace ExpediaRapid.SerializableClasses.Search

    Public Class SearchResponse
        Inherits List(Of PropertyAvailablility)
        Implements IExpediaRapidResponse

        Public Function IsValid(responseString As String, statusCode As Integer) As Boolean Implements IExpediaRapidResponse.IsValid

            If Not String.IsNullOrWhiteSpace(responseString) Then
                Dim token As JToken = JToken.Parse(responseString)
                Try

                    JsonConvert.DeserializeObject(Of SearchResponse)(responseString)
                    Return True

                Catch ex As Exception
                    Return False
                End Try
            End If
            Return False
        End Function
    End Class

End Namespace